satpy/readers/viirs_edr.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022-2023 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""VIIRS NOAA enterprise EDR product reader.
+
+This module defines the :class:`VIIRSJRRFileHandler` file handler, to
+be used for reading VIIRS EDR products generated by the NOAA enterprise
+suite, which are downloadable via NOAA CLASS or on NOAA's AWS buckets.
+
+A wide variety of such products exist and, at present, only a subset are supported.
+
+ - Cloud mask: JRR-CloudMask_v2r3_j01_s202112250807275_e202112250808520_c202112250837300.nc
+ - Cloud products: JRR-CloudHeight_v2r3_j01_s202112250807275_e202112250808520_c202112250837300.nc
+ - Aerosol detection: JRR-ADP_v2r3_j01_s202112250807275_e202112250808520_c202112250839550.nc
+ - Aerosol optical depth: JRR-AOD_v2r3_j01_s202112250807275_e202112250808520_c202112250839550.nc
+ - Surface reflectance: SurfRefl_v1r1_j01_s202112250807275_e202112250808520_c202112250845080.nc
+ - Land Surface Temperature: LST_v2r0_npp_s202307241724558_e202307241726200_c202307241854058.nc
+
+All products use the same base reader ``viirs_edr`` and can be read through satpy with::
+
+    import satpy
+    import glob
+
+    filenames = glob.glob('JRR-ADP*.nc')
+    scene = satpy.Scene(filenames, reader='viirs_edr')
+    scene.load(['smoke_concentration'])
+
+.. note::
+
+    Multiple products contain datasets with the same name! For example, both the cloud mask
+    and aerosol detection files contain a cloud mask, but these are not identical.
+    For clarity, the aerosol file cloudmask is named `cloud_mask_adp` in this reader.
+
+"""
+
+
+import logging
+
+import numpy as np
+import xarray as xr
+
+from satpy import DataID
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.utils import get_chunk_size_limit
+
+LOG = logging.getLogger(__name__)
+M_COLS = 3200
```

Comment:
Is this guaranteed to be constant? The documentation says a granule "typically" has this many columns.

Reply:
I've never seen anything different, but I also wasn't super comfortable with this as my main check for resolution either. I couldn't think of any other way to do it considering some files only have "Columns" and "Rows" dimensions *and* we currently rename the dimensions right away anyway.

Reply:
Ok, fair enough!